src/ir/attrs.cc

```diff
@@ -37,7 +37,7 @@ void DictAttrsNode::InitByPackedArgs(const runtime::TVMArgs& args, bool allow_un
     runtime::TVMArgValue val = args[i + 1];
     if (val.IsObjectRef<ObjectRef>()) {
       dict.Set(key, val.operator ObjectRef());
-    } else if (val.type_code() == kTVMStr) {
+    } else if (String::CanConvertFrom(val)) {
```

Comment:
@junrushao1994 : Sorry to bug you on this. I was wondering whether, the first if condition is always true, when the val is String objref type. Please check once. Thanks!

Reply:
Thank you for your question! Hmmmm but I am not sure I understand if correctly. Just to confirm, "The first if condition", are you referring to "val.type_code() == kTVMStr" in `CanConvertFrom`? Thank you!

Reply:
The Python frontend is still able to pass `runtime::String` back to C++ via packed function. One case I found in our testcases is: https://github.com/apache/incubator-tvm/pull/5687/files#diff-5243f3909dfa484d85d3d8471f259169R422-R424

Reply:
> Thank you for your question! Hmmmm but I am not sure I understand if correctly. Just to confirm, "The first if condition", are you referring to "val.type_code() == kTVMStr" in `CanConvertFrom`? Thank you!
Sorry for confusion.
The first if condition --> if (val.IsObjectRef<ObjectRef>())

Reply:
> The Python frontend is still able to pass `runtime::String` back to C++ via packed function. One case I found in our testcases is: https://github.com/apache/incubator-tvm/pull/5687/files#diff-5243f3909dfa484d85d3d8471f259169R422-R424


I think that may be because of implicit conversion of String(Overloaded) to std::string!

Reply:
I think this is okay because `IsObjectRef` doesn't check the container type for `String`.

https://github.com/apache/incubator-tvm/blob/4347b41a5e64a2a453297b371232d6e101051b3c/include/tvm/runtime/packed_func.h#L1301

Reply:
Yah I got what you mean, and thank you for the clarification @ANSHUMAN87! Totally agree with your points, but we want to unify the way we check "FFI -> string" conversion, so that future users won't end up messing them up. This is the reason why I made this change: After this PR, the only places that use raw kTVMStr are RPC and C-API.

Thank you again for the comments @zhiics @ANSHUMAN87!